awaitility/src/main/java/com/jayway/awaitility/Duration.java

```diff
@@ -331,6 +331,11 @@ public String toString() {
                 '}';
     }
 
+    public int compareTo(Duration other) {
+        if (other == null) return 1;
+        return Long.compare(getValueInMS(), other.getValueInMS());
```

Comment:
I did mention that you're using an API that was introduced in java standard library since version 1.6 but this project is still has its source compatibility level with java 1.5. And this is not the only place where awaitility uses the API greater than its targeted API. IMO this is an issue and awaitility should be either promoted to use java 1.6 target compatibility or do not use later APIs.